CHANGELOG.md

```diff
@@ -7,6 +7,43 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Deprecations
+
+* `jj checkout` and `jj merge` are both deprecated; use `jj new` instead to
+  replace both of these commands in all instances.
+
+  **Rationale**: `jj checkout` and `jj merge` implement two distinct pieces of
```

Comment:
Optional: I feel like this is not a great place to have a detailed rationale. Perhaps you could just put it into the PR description, and say "See <link> for rationale"? I don't insist on this.

Reply:
You could instead say 

> you can alias "co", "checkout", and/or "merge" to "new" if you prefer the old behavior.

Reply:
I just realized this is already discussed in https://github.com/martinvonz/jj/pull/2842#discussion_r1457848112.